app/scripts/metamask-controller.js

```diff
@@ -2239,13 +2236,8 @@ export default class MetamaskController extends EventEmitter {
           checksummedAccountAddress
         ].filter((asset) => {
           if (asset.isERC721 === undefined) {
-            // since the token.address from allTokens is checksumaddress
-            // asset.address have to be changed to lowercase when we are using dynamic list
-            const address = useTokenDetection
-              ? asset.address.toLowerCase()
-              : asset.address;
             // the tokenList will be holding only erc20 tokens
-            if (tokenList[address] !== undefined) {
+            if (tokenList[asset.address.toLowerCase()] !== undefined) {
```

Comment:
Is it ever possible for asset.address to be undefined?

Reply:
Chances are rare, but to be on the safe side, I will add a conditional check; I am also updating the tokenList to use the static list as well, which I missed earlier.